optuna/storages/_journal/storage.py

```diff
@@ -0,0 +1,604 @@
+        trial_id = log["trial_id"]
+
+        if self._trial_exists_and_updatable(trial_id, log):
+            trial = copy.copy(self._trials[trial_id])
+            trial.intermediate_values = {
+                **copy.copy(trial.intermediate_values),
+                log["step"]: log["intermediate_value"],
+            }
+            self._trials[trial_id] = trial
+
+    def _apply_set_trial_user_attr(self, log: Dict[str, Any]) -> None:
+        trial_id = log["trial_id"]
+
+        if self._trial_exists_and_updatable(trial_id, log):
+            assert len(log["user_attr"]) == 1
+            trial = copy.copy(self._trials[trial_id])
+            trial.user_attrs = {**copy.copy(trial.user_attrs), **log["user_attr"]}
+            self._trials[trial_id] = trial
+
+    def _apply_set_trial_system_attr(self, log: Dict[str, Any]) -> None:
+        trial_id = log["trial_id"]
+
+        if self._trial_exists_and_updatable(trial_id, log):
+            assert len(log["system_attr"]) == 1
+            trial = copy.copy(self._trials[trial_id])
+            trial.system_attrs = {
+                **copy.copy(trial.system_attrs),
+                **log["system_attr"],
+            }
+            self._trials[trial_id] = trial
+
+    def _apply_log(self, log: Dict[str, Any]) -> None:
+        op = log["op_code"]
+        if op == JournalOperation.CREATE_STUDY:
+            self._apply_create_study(log)
+        elif op == JournalOperation.DELETE_STUDY:
+            self._apply_delete_study(log)
+        elif op == JournalOperation.SET_STUDY_USER_ATTR:
+            self._apply_set_study_user_attr(log)
+        elif op == JournalOperation.SET_STUDY_SYSTEM_ATTR:
+            self._apply_set_study_system_attr(log)
+        elif op == JournalOperation.SET_STUDY_DIRECTIONS:
+            self._apply_set_study_directions(log)
+        elif op == JournalOperation.CREATE_TRIAL:
+            self._apply_create_trial(log)
+        elif op == JournalOperation.SET_TRIAL_PARAM:
+            self._apply_set_trial_param(log)
+        elif op == JournalOperation.SET_TRIAL_STATE_VALUES:
+            self._apply_set_trial_state_values(log)
+        elif op == JournalOperation.SET_TRIAL_INTERMEDIATE_VALUE:
+            self._apply_set_trial_intermediate_value(log)
+        elif op == JournalOperation.SET_TRIAL_USER_ATTR:
+            self._apply_set_trial_user_attr(log)
+        elif op == JournalOperation.SET_TRIAL_SYSTEM_ATTR:
+            self._apply_set_trial_system_attr(log)
+        else:
+            raise RuntimeError("No corresponding log operation to op_code:{}".format(op))
+
+    def _sync_with_backend(self) -> None:
+        logs = self._backend.read_logs(self._log_number_read)
+        for log in logs:
+            self._log_number_read += 1
+            self._apply_log(log)
+
+    def _create_unique_study_name(self) -> str:
+        self._study_name_suffix_num += 1
+        return "no-name-" + self._pid + "-" + str(self._study_name_suffix_num)
```

Comment:
[question] The naming convention for a study is different from the [InMemoryStorage](https://github.com/optuna/optuna/blob/master/optuna/storages/_in_memory.py#L65). Did you do it on purpose?

Reply:
Also, better to use `DEFAULT_STUDY_NAME_PREFIX `.

Reply:
I had no special reason for this. Will push a fix.